src/Features/Core/Portable/ExternalAccess/UnitTesting/API/NewUnitTestingIncrementalAnalyzerProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.SolutionCrawler;
+using Microsoft.CodeAnalysis.Host;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.Api;
+
+internal sealed partial class NewUnitTestingIncrementalAnalyzerProvider : IUnitTestingIncrementalAnalyzerProvider
+{
+    private readonly string? _workspaceKind;
+    private readonly SolutionServices _services;
+    private readonly INewUnitTestingIncrementalAnalyzerProviderImplementation _incrementalAnalyzerProvider;
+
+    private IUnitTestingIncrementalAnalyzer? _lazyAnalyzer;
+
+    private NewUnitTestingIncrementalAnalyzerProvider(
+        string? workspaceKind,
+        SolutionServices services,
+        INewUnitTestingIncrementalAnalyzerProviderImplementation incrementalAnalyzerProvider)
+    {
+        _workspaceKind = workspaceKind;
+        _services = services;
+        _incrementalAnalyzerProvider = incrementalAnalyzerProvider;
+    }
+
+    // NOTE: We're currently expecting the analyzer to be singleton, so that
+    //       analyzers returned when calling this method twice would pass a reference equality check.
+    //       One instance should be created by SolutionCrawler, another one by us, when calling the
+    //       UnitTestingSolutionCrawlerServiceAccessor.Reanalyze method.
+    public IUnitTestingIncrementalAnalyzer CreateIncrementalAnalyzer()
+        => _lazyAnalyzer ??= new NewUnitTestingIncrementalAnalyzer(_incrementalAnalyzerProvider.CreateIncrementalAnalyzer());
+
+    public void Reanalyze()
+    {
+        var solutionCrawlerService = _services.GetService<IUnitTestingSolutionCrawlerService>();
```

Comment:
I am assuming the indirection from *NewUnitTestingSolutionCrawler* -> *UnitTestingSolutionCrawler* will also be removed in future PRs?

Reply:
Ah I guess `*UnitTestingSolutionCrawler*` is now no longer a thin wrapper but the actual solution crawler. If so, please ignore this question.

Reply:
not exactly sure what you mean.  The expected final place we land is:

1. There is the UnitTestingSolutionCrawler.  It is owned by you guys and lives in the EA layer code we both share.  It is a fork of the normal SolutionCrawler, though with anything you're not currently using/needing stripped out.
2. There is INewUnitTestingXXX.  This is the EA api that allows you to get your existing analyzer to be loaded into the UnitTestingSolutionCrawler and not hte real SolutionCrawler
3. The real SOlutionCrawler goes away.
4. Existing, relevant, SolutionCrawler tests are ported to test UnitTestingSolutionCrawler (but will be owned by you).

Reply:
My bad. I was confused by the naming and misunderstood that the *new* EA layer being introduced in this PR talks to the *old* EA layer which in turn talks to the crawler. Thanks for clarifying that the *new* EA layer talks directly to the crawler (which is now called UnitTestingSolutionCrawler).